### PR TITLE
kvm-xfstests environment improvements

### PR DESCRIPTION
**Pull request by dmonakhov**
Hi,

Please review my patchset. It's primary idea is to improve configuration flexibility.
Note: last cleanup patch allow to reduce kvm-xfstests docker environment from 300Mb to 174M so docker runtime environment becomes closer to optimum size.

### CODE REVIEW

**Comment by tytso** [comment]
None of these commits have a signed-off-by.  Can you rebase with signed-off-by, or send a note here giving me permission to add your signed-off-by?

**Comment by dmonakhov** [comment]
Ok. updated against master and add sign-offs

**Comment by tytso** [comment]
Regarding the last commit (the one that changes kvm-xfstests/Dockerfile to use Alpine).   

Does the usage command really work?  Last I tried running qemu in Docker, it requires --privileged, and it was a gaping security hole.  So it never really seemed to be viable for anything more than as a party trick, to be honest.

As opposed the Dockerfile in the top-level, which actually is useful for doing test builds, and also as a way of testing the file system in whatever Docker environment provides, and which does not require any special privileges.

**Comment by dmonakhov** [comment]
Hm... In fact all it needs is to grant  access to /dev/kvm so 'docker --device /dev/kvm:/dev/kvm' should be enough 

That's depends of your environment. If you try to run in on public cloud then yes, this is security hole.
But it you run in on your cluster which you (or your team) own  then this is very usefull. For example I've join old crappy boxes to one kubernetes cluster which aggregates 300 vcpu and 1Tbmem . I split xfstests to ~100 independent jobs and run it in paralles as kubernetes-jobs. This allow me to complete 'smoke' scenario ~3min.  IMHO this is fantastic achievement which may be usefull for others.

**Comment by dmonakhov** [comment]
Yeah. It seems that running qemu-kvm inside container is now safe in GCE :) https://cloudplatform.googleblog.com/2017/09/introducing-nested-virtualization-for.html
https://cloud.google.com/compute/docs/instances/enable-nested-virtualization-vm-instances

**Comment by tytso** [comment]
Sure, but nested virtualization doesn't buy us much over simply running xfstests in GCE natively.  That's what gce-xfstests is all about.    In fact, thanks to some great work by a summer intern, we now have now have test manager so that we can break up the work to muliple VM's, monitor the VM's to see if they wedge, capture the serial console output, etc.   At the moment we are only breaking up the work so that it's one file system config per VM, but it allows us to collapse 23 hours (from running "gce-xfstests -c ext4/all -g auto") down to 3 hours.    We have thought about breaking up the work in a more finer-grained way, but there are some tradeoffs.  In particular, if we shard in a too-fine-grained way, then lockdep bugs and memory bugs are much less likely to be detected.    That might be OK for a smoke test, but for a full regression test, I'd want to stick with one VM per file system config.

Back to the original question.  If the only thing needed is "kvm-xfstests -p /dev/kmem:/dev/kmem", shouldn't that be mentioned in the usage message listed in the Dockerfile?

**Review by amir73il — COMMENTED** [review]
typo in subject s/eraly/early

File: kvm-xfstests/config.kvm [hunk 1]
```diff
@@ -40,3 +41,11 @@ MONITOR="-monitor telnet:localhost:7498,server,nowait"
 # interception entirely.
 CONSOLE=" -serial mon:stdio"
 #CONSOLE=" -chardev stdio,id=console,signal=off -serial chardev:console"
+
+# Lost IOs have been observed with the default aio=threads with
+# QEMU 2.5.0 when running crash consistency tests.
+# The issue has not been observed with QEMU 2.9.0.
+# Native AIO is probably better for our goal which is test proofing
+# production systems.
+AIO=",aio=native"
```
Comment (amir73il) [inline kvm-xfstests/config.kvm]: minor suggestion, maybe keep this up near CACHE and the disks configuration?

**Comment by dmonakhov** [comment]
Ok. I've got what you are talking about. Yes, usage definitely need to be updated. See updated pathes

File: kvm-xfstests/config.kvm (hunk 1 above)
Comment (dmonakhov) [reply kvm-xfstests/config.kvm]: Reasonable. Thank you.

**Comment by dmonakhov** [comment]
I've add one more patch which adds travis-ci script. I've copy-pasted it from fio.git . It is very trivial but still usefull. Here is example of build-log https://travis-ci.org/dmonakhov/xfstests-bld/builds/281336943

**Comment by tytso** [comment]
OK, I've integrated all of your changes into my tree.

Note that Travis CI is requires that you give them read:org privileges, which means they get to see basic information about all of your organization's private repositories.   Your company have rules which absolutely prohibit leakage of the companies private informaiton to a third party, such as Travis CI.   At least for me, doing continuous integration *after* I've made the git repo published for all of the world to see doesn't actually help all that much.   So I just tend to do a full build *plus* trying to run a few smoke tests before I push changes out publically --- and so Travis doesn't actually buy me much, personally.

**Pull request closed without merging**